provider/server/machine_delete_test.go

```diff
@@ -80,7 +184,30 @@ var _ = Describe("DeleteMachine", func() {
 		})
 		Expect(err).NotTo(HaveOccurred())
 
-		By("ensuring machine is deleted")
+		By("ensuring machine is in Terminating state after delete")
+		Eventually(func() iri.MachineState {
+			listResp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
+				Filter: &iri.MachineFilter{
+					Id: createResp.Machine.Metadata.Id,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(listResp.Machines).Should(HaveLen(1))
+			return listResp.Machines[0].Status.State
+		}).Should(Equal(iri.MachineState_MACHINE_TERMINATING))
+
+		By("ensuring machine is not deleted until gracefulShutdownTimeout ")
+		Eventually(func() []*iri.Machine {
+			listResp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
+				Filter: &iri.MachineFilter{
+					Id: createResp.Machine.Metadata.Id,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			return listResp.Machines
+		}).Within(gracefulShutdownTimeout).ProbeEvery(5).Should(HaveLen(1))
```

Comment:
i don't think this condition is correct. During graceful period you will get machineList with one machine, so eventually will do only 1 loop.